working/augmentation-libraries/parts_with_imports.md

```diff
@@ -296,6 +296,118 @@ library, so the library author should fix the conflict by renaming the prefix.
+> * The extension is declared by the library of the Dart file, in which
+>   case it's accessible with priority zero.
+> * The extension is accessible by import with priority *n* in the Dart file.
+>
+> An extension is accessible by import with priority *n* in a Dart file iff:
+> * The Dart file *imports* the extension, meaning that the Dart file
+>   contains a non-deferred import directive whose imported
+>   library has the extension in its export scope, and where the name of
+>   the extension is not private, and is not hidden by either a `hide`
+>   combinator mentioning the extension name or by a `show` combinator
+>   not mentioning the name.
+>   In that case the extension is accessible with priority 1.
+>   _This includes non-deferred imports with a prefix, and even an unnamed
+>   prefix using the the "wildcard variable" unnamed variable notation
+>   `as _`._
+> * Or otherwise, if the Dart file is a part file and the extension is
+>   accessible by import in the part file's parent file with priority *n*.
+>   In that case the extension is accessible by import to the current file
+>   with priority *n + 1*. _(Lower is better.)_
+
+The "priority *n*" of an extension for a particular Dart file is a measure
+of how far up in the parent chain the import for that extension occurred.
+A value of "1" means the current Dart file, and each step up the parent
+chain increases the value by one. A value of zero means the declaration
+is in the current library, and that is the priority of the extension
+in all files of the library, even if the same extension is *also* imported.
+
+[extension feature specification on accessibility]: https://github.com/dart-lang/language/blob/main/accepted/2.7/static-extension-methods/feature-specification.md#accessibility "Accessibility"
+
+The rules for *applicability* of extensions do not change.
+
+The conflict resolution, used when multiple accessible extensions apply
+to an invocation, are updated to give priority to an extension imported
+by a part file over extensions made accssible by its parent file.
+This maintains the guarantee that a part file can ignore
+parent file imports, as long as it imports all its own dependencies:
+every name it refers to itself, *and* every extension that it refers
+to *implicitly*.
+
+The [extension feature specification on specificity] is updated to
+the following:
+
+> Let *i* be a member invocation with target expression `e` and corresponding
+> member name *m*, and let *E1* and *E2* denote different accessible and
+> applicable extensions for *i*.
+> 
+> Whether *E1* is more specific than *E2* wrt. *i* of *e* is determined by
+> the following steps, stopping when an answer is given:
+> * If *E1* is accessible with priority *n1* &gt; 0 and *E2* is accessible
+>   with priority *n2* &gt; 0 _(so both accessible by import)_ then
+>     * if *n1* < *n2* then *E1* is more specific than *E2*, and
+>     * if *n2* < *n1* then *E1* is *not* more specific than *E2*.
```

Comment:
Is there a reason this is phrased as "E1 is not more specific than E2" instead of saying "E2 is more specific than E1"?

It reads to me like you're trying to define a partial order, but I don't quite see why when `n2 < n1` in this clause. I might be missing some context.

Reply:
It is a partial order. And even more, this text is *part* of a partial order specification.
There can be distinct extensions where neither is more specific than the other. If they are in the set of applicable extensions, then lookup fails.

Saying that *E2* is more specific than *E1* would not answer whether *E1* is more specific than *E2* or not. 
That would be assuming that the relation is anti-symmetric. (It probably is, but that's a consequence of the definition deliberately avoiding any mutual more-specific relations, and this rule is part of making that so, so it can't also assume it.)

The relation being defined is "*E1* more specific than *E2*" which should give an answer of true or false.
The items are in priority order, and this item gives an answer of "yes" in one case and "no" in another, and otherwise it falls through to the next item (hence "part of a partial order").
Answering any other question doesn't help. (But the rules do imply that if *E1* is not more specific than *E2* *for this reason*, then *E2* is more specific than *E1*, since it's the same *n* values.)